### ISSUE TITLE
Fatal Error: Cannot access empty property

### ISSUE THREAD

**Issue by Beee4life** [timber/timber#1269]
My template worked without any problem.... I then updated to 1.1.11 and got this error. I'm not sure if the update caused this, but I first noticed it after the update.

`Fatal error: Cannot access empty property in /Users/Beee/HTML/project-folder/webapp/content/plugins/timber-library/lib/Image.php on line 285`

Line 285 is the second line in this statement:

    foreach ( $custom as $key => $value ) {
        $this->$key = $value[0];
    }

This is the code that messes it up because if I uncomment it, it works (but no image).

    {% for item in updated %}
        <li class="posttype__list-item">
            {% set image = TimberImage( item.get_field( 'sd_profile_photo' ) ) %}
            {% include 'teasers/posttype.twig' with {
                'post' : item,
                'image' : image
            } %}
        </li>
    {% endfor %}

if I set it like this, the image_id returns the proper ID, but when I feed it into TimberImage it breaks again.

    {% set image_id = item.get_field( 'sd_profile_photo' ) %}
    {% set image = TimberImage( image_id) %}

If I add an hardcoded image ID into it, it also still breaks.

I use: 
* WP 4.7
* Timber 1.11.1

It's installed through a composer file.

I have no idea what could have caused this, ~~because on a single template, it still works~~. but when I list an image in a get_posts loop, it won't show.

**Comment by Beee4life** [comment]
Further investigating learns if I dump $key and $value they are both not empty.
If I dump $this the ID shows NULL
**I don't think this should be NULL**

$iid is properly passed on line 267 because if I dump it I get the expected ID and if I dump $image_info I can find the image ID in there. See the output http://pastebin.com/AWbbNHXC.

I then tried dumping $this->ID on line 283 which also produces the expected ID.

And lastly I discovered this error, hiding behind my nav menu.

`Strict Standards:  Declaration of Timber\Image::init() should be compatible with Timber\Post::init($pid = false) in /Users/Beee/HTML/project-folder/webapp/content/plugins/timber-library/lib/Image.php on line 48`

**Comment by Beee4life** [comment]
I just did a clean install and only activated timber which gave the error as mentioned above and I saw a new warning, which I didn't notice before since the plugin was already active.

`The plugin generated 239 characters of unexpected output during activation. If you notice “headers already sent” messages, problems with syndication feeds or other issues, try deactivating or removing this plugin.`

**Comment by Beee4life** [comment]
Another site which also runs Timber which hadn't been worked in over more than a month was also updated to Timber and broke completely as well.

I deactivated w3tc after several attempts to fix it by downgrading/upgrading and that appeared to have worked. Can it be it clashes with w3tc in some way ????

Although this can't be entirely it because on my other site (for which I started this issue) I don't have w3tc because it's still in dev.

**Comment by jarednova** [comment]
thanks @Beee4life — I'll look into this!

**Comment by Beee4life** [comment]
some more info I just discovered.... if I would feed an URL into then it does work, but then it doesn't show the proper format image.... I return an image ID from an ACF image upload field.... but the full image is not the image i'd like to show since it's a cropped version which is 'determined' after this.

**Comment by Beee4life** [comment]
if you want i can give you access to a staging ftp which is a copy of dev...

**Comment by Beee4life** [comment]
Just found out it also interferes with deleting a custom post type. I get the same fatal error. If I delete timber and then delete it works fine.

**Comment by Beee4life** [comment]
i played around some more with the definition of how the image ID was called because I saw an image appear somewhere on the site.

The notation there was different.
On the frontpage I used `{% set image = TimberImage(item.get_field( 'sd_profile_photo' )) %}` where as on that other page I used `{% set image = TimberImage(item.sd_profile_photo) %}`.

It appears this seems to fix my issue for now... will continue to investigate and report any findings.
